### PR TITLE
Denormalised, added models and admin

### PR DESCRIPTION
**Pull request by edanweis**
Couldn't really re-use the existing Entry model via foreign key because I denormalised the data differently. One thing I found confusing is whether to create models quickly and easily which suit django-admin, or properly denormalise models and deal wth work-arounds for django-admin, eg: listing many-many relationships, and editing multiple models in a single admin form, not easy for a beginner with Django and RDBMS. Mongoengine now looks appealing.

### CODE REVIEW

File: timetracker/entries/admin.py [hunk 1]
```diff
@@ -3,10 +3,25 @@
 # Register your models here.
 from . import models
 
-class EntryAdmin(admin.ModelAdmin):
-    list_display = ('project', 'description', 'start', 'stop')
-    list_filter = ('project',)
-    search_fields = ('project', 'description',)
+# Django admin site doesn't create forms using attributes from differnet models, so each model must be edited seperately (!?)
```
Comment (sesh) [inline timetracker/entries/admin.py:6]: Nice pick up! I think we'll cover it in Lesson 6 (Advanced Topics), but if you're interested you should check out the documentation on Admin Inlines. You create them in a _very_ similar way to what you're done here.

https://docs.djangoproject.com/en/1.8/ref/contrib/admin/#inlinemodeladmin-objects

File: timetracker/entries/models.py [hunk 2]
```diff
@@ -2,24 +2,39 @@
...
+
+    #how to create a method for displaying foreign key attributes? eg: a client's projects?
+
+    def __str__(self):
+        return '{}'.format(self.name)
 
 class Project(models.Model):
-    client = models.CharField(max_length=200)
     name = models.CharField(max_length=200)
+    client = models.ManyToManyField(Client, related_name="+")
```
Comment (sesh) [inline timetracker/entries/models.py:17]: You shouldn't set `related_name="+"` here. You don't actually need a related_name at all in this instances.

Settings it to `+` stops Django from creating the reverse relationship between `Client` and `Project`, which is the problem you've highlighted with your comment in the `Client` model.

File: timetracker/entries/models.py [hunk 3]
```diff
@@ -2,24 +2,39 @@
...
-    project = models.ForeignKey('Project')
     description = models.CharField(max_length=200)
-
-    def __str__(self):
-        return '[{} - {}] ({}) {}'.format(self.start, self.stop, self.project, self.description)
-
+    project = models.ManyToManyField(Project, related_name="+", blank=True)
     def is_finished(self):
         return self.stop is not None
+    is_finished.boolean = True
```
Comment (sesh) [inline timetracker/entries/models.py:34]: What's the purpose of setting this here?

**Comment by sesh** [comment]
Nice work @edanweis. I like the idea of splitting out in `Task` instead of `Entry`.

File: timetracker/entries/models.py (hunk 3 above)
Comment (edanweis) [inline timetracker/entries/models.py:34]: I forgot how to get icons showing in admin.. this is working. Maybe should be in admin.py?

File: timetracker/entries/models.py (hunk 2 above)
Comment (edanweis) [inline timetracker/entries/models.py:17]: I get a Reverse query name clash if I add a foreignkey to the Project, so I wasn't sure how to do this. Hopefuly follow up in class.

**Pull request closed without merging**